buildSrc/src/main/groovy/com/uber/okbuck/generator/BuckFileGenerator.groovy

```diff
@@ -97,7 +94,7 @@ final class BuckFileGenerator {
                     }
                     break
                 default:
-                    break
+                    throw new IllegalArgumentException("ProjectType not handled: $projectType")
```

Comment:
My goal was to make it easier when adding a new `ProjectType` to figure out all the places that it needs to be added to in the code.

I didn't realize this would crash in other ways, are you saying that if incompatible projects aren't explicitly excluded from `buckProjects` in the `OkbuckExtension` that this method will crash here?

From my experience, incompatible projects will just crash further down the pipeline if they don't already crash here. Kotlin Android projects are one example of this so I don't think it will change much.

I'm happy to add the following code if it would help this problem:
```
UNKOWN:
  break;
default:
  throw new IllegalArgumentException("...");
```